pkg/server/http/option.go

```diff
@@ -83,3 +84,10 @@ func Handler(h http.Handler) Option {
 		o.Handler = h
 	}
 }
+
+// Middlewares provides a function to register middlewares
+func Middlewares(val ...func(handler http.Handler) http.Handler) Option {
+	return func(o *Options) {
+		o.Middlewares = val
+	}
+}
```

Comment:
I think instead of using variadic arguments we rather use `append` and append at the options function, that will allow your service to, on creation, accept a list of middlewares and the overall code is cleaner. But this is perfectly fine as well :)